docs/self-managed/concepts/secondary-storage/configuring-secondary-storage.md

```diff
@@ -0,0 +1,131 @@
+This section explains how to configure secondary storage through different deployment methods and how to disable it when running the Zeebe engine alone.
+
+## Configuration options
+
+You can configure secondary storage using **Helm charts**, **Docker Compose**, or **manual configuration files**.
+
+### Helm
+
+When deploying with Helm, set the secondary storage type and connection details in your `values.yaml` file:
+
+```yaml
+data:
+  secondary-storage:
+    type: rdbms
+    rdbms:
+      url: jdbc:h2:mem:camunda
+      username: sa
+      password:
+```
+
+You can switch to Elasticsearch by using:
+
+```yaml
+data:
+  secondary-storage:
+    type: elasticsearch
+    elasticsearch:
+      url: http://elasticsearch:9200/
+```
+
+To explicitly disable secondary storage (for example, when running only the Zeebe engine), you can set:
+
+```yaml
+global:
+  noSecondaryStorage: true
+```
+
+When this flag is set, all secondary-storage-dependent components are automatically disabled.
+
+### Docker Compose
+
+If you’re using Docker Compose, configure your environment variables within the relevant service definition:
+
+```yaml
+environment:
+  - CAMUNDA_DATA_SECONDARYSTORAGE_TYPE=rdbms
+  - CAMUNDA_DATA_SECONDARYSTORAGE_RDBMS_URL=jdbc:h2:mem:camunda
+  - CAMUNDA_DATA_SECONDARYSTORAGE_RDBMS_USERNAME=sa
+  - CAMUNDA_DATA_SECONDARYSTORAGE_RDBMS_PASSWORD=
+```
+
+For Elasticsearch:
+
+```yaml
+environment:
+  - CAMUNDA_DATA_SECONDSTORAGE_TYPE=elasticsearch
+  - CAMUNDA_DATA_SECONDSTORAGE_ELASTICSEARCH_URL=http://elasticsearch:9200
+```
+
+To disable secondary storage:
+
+```yaml
+environment:
+  - CAMUNDA_DATA_SECONDARYSTORAGE_TYPE=none
+```
+
+### Manual configuration (application.yaml)
+
+In Self-Managed or Camunda 8 Run deployments, you can also configure storage directly in the `application.yaml` file:
+
+```yaml
+data:
+  secondary-storage:
+    type: rdbms
+    rdbms:
+      url: jdbc:h2:file:./camunda-data/h2db
+      username: sa
+      password:
+```
+
+For Elasticsearch:
+
+```yaml
+data:
+  secondary-storage:
+    type: elasticsearch
+    elasticsearch:
+      url: http://localhost:9200/
+```
+
+## Choosing a storage backend
+
+| Scenario                                  | Recommended backend              | Reason                                                        |
+| ----------------------------------------- | -------------------------------- | ------------------------------------------------------------- |
+| Local testing or Camunda 8 Run quickstart | H2                               | Fast, lightweight, and runs entirely in memory or file-based. |
+| Production workloads                      | Elasticsearch or supported RDBMS | Scalable and persistent; designed for concurrent queries.     |
+| Debugging and troubleshooting             | H2 or PostgreSQL                 | Easier to inspect and visualize data.                         |
```

Comment:
I find it a bit weird that we recommend Postgres for this. Can you expand as to what we want to communicate here or what use case we want to cover?

Reply:
Thanks for explaining. I think it's fine to leave it or change it as you proposed, was mostly just confused on my end 🙂